apps/aecore/src/aec_headers.erl

```diff
@@ -368,6 +369,24 @@ encode_block_hash(key, Hash) ->
 encode_block_hash(micro, Hash) ->
     aec_base58c:encode(micro_block_hash, Hash).
 
+-spec deserialize_from_client(key, map()) -> {ok, header()} | {error, term()}.
+deserialize_from_client(key, KeyBlock) ->
+    try
+        {ok, #key_header{height       = maps:get(<<"height">>, KeyBlock),
+                         prev_hash    = decode_block_hash(maps:get(<<"prev_hash">>, KeyBlock)),
```

Comment:
The function decode_block_hash is a nested base58c safe_decode on first key and then micro blocks. It would be better to add a safe_decode(block, Binary) to base58c that does exactly this. Here one could then use that function. NOTE that you should not (as the comment suggests) look at the first two digits to decide which it is, because safe_decode enforces the thing to be correct.